### PR TITLE
[Az.Accounts] Fixed incorrect access token

### PR DESCRIPTION
**Pull request by VeryEarly**
## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Accounts/Authentication/Factories/AuthenticationFactory.cs [hunk 1]
```diff
@@ -475,7 +475,7 @@ private string GetEndpointToken(IAzureAccount account, string targetEndpoint)
             {
                 tokenKey = AzureAccount.Property.KeyVaultAccessToken;
             }
-            if (string.Equals(targetEndpoint, AzureEnvironment.ExtendedEndpoint.MicrosoftGraphEndpointResourceId, StringComparison.OrdinalIgnoreCase))
+            if (string.Equals(targetEndpoint, AzureEnvironment.ExtendedEndpoint.MicrosoftGraphEndpointResourceId, StringComparison.OrdinalIgnoreCase) || string.Equals(targetEndpoint, AzureEnvironment.ExtendedEndpoint.MicrosoftGraphUrl, StringComparison.OrdinalIgnoreCase))
```
Comment (isra-fel) [inline src/Accounts/Authentication/Factories/AuthenticationFactory.cs:478]: Should we do this for key vault as well?

Comment (VeryEarly) [reply src/Accounts/Authentication/Factories/AuthenticationFactory.cs:478]: > Should we do this for key vault as well?

if you mean allow "AzureKeyVaultDnsSuffix" to get keyvault accesstoken I think it's safe to add this key.
But considering this key was passed in by ClientFactory.CreateArmClient, it's less likely the suffix will be used to create service client. Do you have other use case of this method which suffix could be passed in?

Comment (isra-fel) [reply src/Accounts/Authentication/Factories/AuthenticationFactory.cs:478]: In that case I suggest adding a comment reminding others not to follow the practice of MSGraph.

Comment (VeryEarly) [reply src/Accounts/Authentication/Factories/AuthenticationFactory.cs:478]: > In that case I suggest adding a comment reminding others not to follow the practice of MSGraph.

create issue: https://github.com/Azure/azure-powershell/issues/18634 to describe the ideal fix

**Review by isra-fel — APPROVED** [review]

**Pull request merged**